cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -287,22 +298,82 @@ friendlyVotingProcedures
   :: ConwayEraOnwards era -> L.VotingProcedures (ShelleyLedgerEra era) -> Aeson.Value
 friendlyVotingProcedures cOnwards x = conwayEraOnwardsConstraints cOnwards $ toJSON x
 
-redeemerIfShelleyBased :: MonadWarning m => CardanoEra era -> TxBody era -> m [Aeson.Pair]
-redeemerIfShelleyBased era tb = monoidForEraInEonA @ShelleyBasedEra era $
-  \shEra -> do
-    redeemerInfo <- friendlyRedeemer shEra tb
-    return ["redeemers" .= redeemerInfo]
-
-friendlyRedeemer :: MonadWarning m => ShelleyBasedEra era -> TxBody era -> m Aeson.Value
-friendlyRedeemer _ (ShelleyTxBody _ _ _ TxBodyNoScriptData _ _) = return Aeson.Null
-friendlyRedeemer _ (ShelleyTxBody _ _ _ (TxBodyScriptData _ _ r) _ _) = encodingToJSON $ L.toCBOR r
+redeemerIfAlonzoOnwards :: MonadWarning m => CardanoEra era -> TxBody era -> m [Aeson.Pair]
+redeemerIfAlonzoOnwards cea tb = do
+  redeemerInfo <-
+    caseByronOrShelleyBasedEra
+      (return Aeson.Null)
+      ( const $ do
+          let ShelleyTx sbe ledgerTx = makeSignedTransaction [] tb
+          caseShelleyToMaryOrAlonzoEraOnwards
+            (\_ -> return Aeson.Null)
+            (`friendlyRedeemers` ledgerTx)
+            sbe
+      )
+      cea
+  return ["redeemers" .= redeemerInfo]
  where
-  encodingToJSON :: MonadWarning m => Encoding -> m Aeson.Value
-  encodingToJSON e =
-    eitherToWarning Aeson.Null $
-      first ("Error decoding redeemer: " ++) $
-        fromFlatTerm (decodeValue True) $
-          toFlatTerm e
+  friendlyRedeemers
+    :: MonadWarning m
+    => AlonzoEraOnwards era
+    -> Ledger.Tx (ShelleyLedgerEra era)
+    -> m (Aeson.Value)
+  friendlyRedeemers aeo tx =
+    alonzoEraOnwardsConstraints aeo $ do
+      redeemerList <-
+        mapM
```

Comment:
Can you clean this up. It's going to be difficult for new devs to understand what is happening here. Factor out standalone functions if you have to.

Reply:
You are right, there was a lot of room for improvement. Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/861/commits/9e24d1793749c46654b7bce684d78adc6c7834bb